bin/trigger_azure_pipelines.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""A script which triggers and waits for Azure DevOps to complete a build"""
+
+import os
+import sys
+import time
+from ast import parse
+
+from azure.devops.connection import Connection
+from msrest.authentication import BasicAuthentication
+
+# https://learn.microsoft.com/en-us/rest/api/azure/devops/build/builds/get?view=azure-devops-rest-7.1#buildresult
+FAILED_STATUSES: set[str] = {"canceled", "partiallySucceeded", "failed"}
+COMPLETED_STATUSES: set[str] = FAILED_STATUSES | {"succeeded"}
+
+
+def create_ado_connection(organization, ado_pat) -> Connection:
+    """Returns an ADO connection to call the ADO REST APIs."""
+
+    return Connection(
+        base_url=f"https://dev.azure.com/{organization}",
+        creds=BasicAuthentication("", ado_pat),
+    )
+
+
+def build_status_match(build, expected_status_list):
+    """Returns True if status of all the environments of the build match any of the expected statuses"""
+    # Each build can have one or more environments (stages).
+    build_env_statuses = [environment.status for environment in build.environments]
+    return all(status in expected_status_list for status in build_env_statuses)
+
+
+def wait_for_build(build_client, project, build_id):
+    """Wait for the azure devops build to finish"""
+    build = build_client.get_build(project, build_id)
+
+    # polling interval is set in accordance with the rate limits specified here:
+    # https://learn.microsoft.com/en-us/azure/devops/integrate/concepts/rate-limits?view=azure-devops
+    polling_interval_seconds = 30
+
+    # Wait until the build have a complete status.
+    while build.result not in COMPLETED_STATUSES:
+        time.sleep(polling_interval_seconds)
+        build = build_client.get_build(project, build_id)

```

Comment:
Not at all a blocker but just curious - Could there be any scenario of a hung build and this script waits till the build is done?

Reply:
Good question, Haard! There is a note I have added in the description about the same scenario.  
There is a timeout set for OE build pipelines. So, the build once it's triggered, ADO enforces a timeout. This make sures we don't wait forever on the build. 
However, if the build hasn't started and is waiting in the queue to get started, there is no timeout while waiting in the queue. In such case, we can manually cancel the ADO build and it will not have any impact because it hasn't started.